xarray/core/computation.py

```diff
@@ -579,28 +599,61 @@ def apply_variable_ufunc(
                 "``.load()`` or ``.compute()``"
             )
         elif dask == "parallelized":
-            input_dims = [broadcast_dims + dims for dims in signature.input_core_dims]
             numpy_func = func
 
+            if dask_gufunc_kwargs is None:
+                dask_gufunc_kwargs = {}
+
+            output_sizes = dask_gufunc_kwargs.pop("output_sizes", dict())
+            if output_sizes:
+                output_sizes_renamed = dict()
+                for key, value in output_sizes.items():
+                    if key not in signature.all_output_core_dims:
+                        raise ValueError(
+                            f"dimension name '{key}' in 'output_sizes' must correspond to output_core_dims"
+                        )
+                    output_sizes_renamed[signature.dims_map[key]] = value
```

Comment:
this part could be moved into a helper function and called in a dict comprehension, but I guess it's also fine to leave it as is (IIRC dictionary comprehensions tend to be faster than assigning lots of keys to a existing dictionary, but in this case there's usually not enough keys for this to really make a difference).